engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateByOperatorFactory.java

```diff
@@ -219,10 +219,9 @@ public int hashKey(final ColumnUpdateOperation clause) {
 
                         final RollingOpSpec rollingSpec = (RollingOpSpec) spec;
 
-                        // Windowed ops are unique per type (ticks/time-based) and window dimensions
-                        hash = 31 * hash + Objects.hashCode(rollingSpec.revWindowScale().timestampCol());
-                        hash = 31 * hash + Long.hashCode(rollingSpec.revWindowScale().timescaleUnits());
-                        hash = 31 * hash + Long.hashCode(rollingSpec.fwdWindowScale().timescaleUnits());
+                        // Leverage the immutable hashCode() function.
+                        hash = 31 * hash + Objects.hashCode(rollingSpec.revWindowScale());
+                        hash = 31 * hash + Objects.hashCode(rollingSpec.fwdWindowScale());
```

Comment:
`WindowScale` is abstract and does not define an explict `hashCode` function.  I assume hashing is taken into account by `ImmutableWindowScale`, if I infer your comment correctly.  This comment is just a sanity check, since I was not an original reviewer.

Reply:
Yes, `ImmutableWindowScale` provides both `hashCode()` and `equals()` functions.